Mathlib/Algebra/Homology/ShortComplex/Homology.lean

```diff
@@ -33,22 +33,171 @@ namespace CategoryTheory
+    LeftHomologyMapData.commπ_assoc, HomologyData.comm, LeftHomologyMapData.commi_assoc,
+    RightHomologyMapData.commι, HomologyData.comm_assoc, RightHomologyMapData.commp]
+
+instance : Subsingleton (HomologyMapData φ h₁ h₂) := ⟨by
+  rintro ⟨left₁, right₁⟩ ⟨left₂, right₂⟩
+  simp only [mk.injEq, eq_iff_true_of_subsingleton, and_self]⟩
+
+instance : Inhabited (HomologyMapData φ h₁ h₂) :=
+  ⟨⟨default, default⟩⟩
+
+instance : Unique (HomologyMapData φ h₁ h₂) := Unique.mk' _
+
+variable (φ h₁ h₂)
+
+/-- A choice of the (unique) homology map data associated with a morphism
+`φ : S₁ ⟶ S₂` where both short complexes `S₁` and `S₂` are equipped with
+homology data. -/
+def homologyMapData : HomologyMapData φ h₁ h₂ := default
+
+variable {φ h₁ h₂}
+
+lemma congr_left_φH {γ₁ γ₂ : HomologyMapData φ h₁ h₂} (eq : γ₁ = γ₂) :
+  γ₁.left.φH = γ₂.left.φH := by rw [eq]
+
+end HomologyMapData
+
+namespace HomologyData
+
+/-- When the first map `S.f` is zero, this is the homology data on `S` given
+by any limit kernel fork of `S.g` -/
+@[simps]
+def ofIsLimitKernelFork (hf : S.f = 0) (c : KernelFork S.g) (hc : IsLimit c) :
+    S.HomologyData where
+  left := LeftHomologyData.ofIsLimitKernelFork S hf c hc
+  right := RightHomologyData.ofIsLimitKernelFork S hf c hc
+  iso := Iso.refl _
+
+/-- When the first map `S.f` is zero, this is the homology data on `S` given
+by the chosen `kernel S.g` -/
+@[simps]
+noncomputable def ofHasKernel (hf : S.f = 0) [HasKernel S.g] :
+    S.HomologyData where
+  left := LeftHomologyData.ofHasKernel S hf
+  right := RightHomologyData.ofHasKernel S hf
+  iso := Iso.refl _
+
+/-- When the second map `S.g` is zero, this is the homology data on `S` given
+by any colimit cokernel cofork of `S.f` -/
+@[simps]
+def ofIsColimitCokernelCofork (hg : S.g = 0) (c : CokernelCofork S.f) (hc : IsColimit c) :
+    S.HomologyData where
+  left := LeftHomologyData.ofIsColimitCokernelCofork S hg c hc
+  right := RightHomologyData.ofIsColimitCokernelCofork S hg c hc
+  iso := Iso.refl _
+
+/-- When the second map `S.g` is zero, this is the homology data on `S` given by
+the chosen `cokernel S.f` -/
+@[simps]
+noncomputable def ofHasCokernel (hg : S.g = 0) [HasCokernel S.f] :
+    S.HomologyData where
+  left := LeftHomologyData.ofHasCokernel S hg
+  right := RightHomologyData.ofHasCokernel S hg
+  iso := Iso.refl _
+
+/-- When both `S.f` and `S.g` are zero, the middle object `S.X₂` gives a homology data on S -/
+@[simps]
+noncomputable def ofZeros (hf : S.f = 0) (hg : S.g = 0) :
+    S.HomologyData where
+  left := LeftHomologyData.ofZeros S hf hg
+  right := RightHomologyData.ofZeros S hf hg
+  iso := Iso.refl _
+
+/-- If `φ : S₁ ⟶ S₂` is a morphism of short complexes such that `φ.τ₁` is epi, `φ.τ₂` is an iso
+and `φ.τ₃` is mono, then a homology data for `S₁` induces a homology data for `S₂`.
+The inverse construction is `ofEpiOfIsIsoOfMono'`. -/
+@[simps]
+noncomputable def ofEpiOfIsIsoOfMono (φ : S₁ ⟶ S₂) (h : HomologyData S₁)
```

Comment:
Not sure if this is useful, but `ofEpiOfIsIsoOfMono` and `ofEpiOfIsIsoOfMono'` could be bundled together into an equiv.

Reply:
Indeed, I do not see situations where an equality between two `HomologyData` would be useful. Relations between `Left/Right/HomologyData` are better expressed using `HomologyMapData`. (Actually, in early versions of this code, `HomologyMapData` were morphisms in the category of short complexes equipped with a `HomologyData`.)